force-app/main/default/classes/RD2_ValidationService.cls

```diff
@@ -180,32 +180,44 @@ public with sharing class RD2_ValidationService {
 
         Boolean isNew = oldRds == null || oldRds.isEmpty();
 
-        if (isNew) {
-            errorRds = validateOnInsert();
-
-        } else {
-            errorRds = validateOnChange();
-        }
+        errorRds = validateOnInsertOrUpdate(isNew);
 
         return errorRds;
     }
 
     /**
     * @description Validate RD data on insert
+    * @param isNew True if the Recurring Donation is New, False if not
     * @return void
     */
-    public List<ErrorRecord> validateOnInsert() {
+    public List<ErrorRecord> validateOnInsertOrUpdate(Boolean isNew) {
```

Comment:
I think we no longer need a separate function to do the validation because insert and update are now combined into 1. This function can now merge into `validate()` function.

Reply:
This is true!